### PR TITLE
[*] feat: textWrap

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/editor/web/src/RichContentEditor/handleBackspaceCommand.js [hunk 1]
```diff
@@ -23,8 +25,15 @@ export default editorState => {
   const content = editorState.getCurrentContent();
   const startKey = selection.getStartKey();
   const blockBefore = content.getBlockBefore(startKey);
+  const cursorAtStartOfBlock = selection.isCollapsed() && selection.getAnchorOffset() === 0;
 
   if (blockBefore && blockBefore.getType() === 'atomic') {
+    if (cursorAtStartOfBlock) {
+      const blockBeforeSelection = SelectionState.createEmpty(blockBefore.getKey());
+      setNativeSelectionToBlock(blockBefore);
+      return EditorState.forceSelection(editorState, blockBeforeSelection);
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/handleBackspaceCommand.js:34]: return setSelectionToBlock(blockBefore, editorState)

File: packages/viewer/web/src/RichContentViewer.tsx [hunk 2]
```diff
@@ -76,6 +76,7 @@ export interface RichContentViewerProps {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   setRef?: React.RefObject<any>;
   onMouseOver?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+  textWrap: boolean;
```
Comment (shaulgo) [inline packages/viewer/web/src/RichContentViewer.tsx:79]: revert

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/common/web/src/Utils/classNameStrategies.ts [hunk 3]
```diff
@@ -13,7 +14,7 @@ export const alignmentClassName: ClassNameStrategy = (componentData, theme, styl
 
 export const sizeClassName: ClassNameStrategy = (componentData, theme, styles) => {
   const { size } = componentData.config || {};
-  if (!size) {
+  if (!size || componentData.type === DIVIDER_TYPE) {
```
Comment (shaulgo) [inline packages/common/web/src/Utils/classNameStrategies.ts:17]: make sizeClassName strategy for divider (in plugin)

File: packages/common/web/src/types/commonTypes.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export type RichContentTheme = {
 };
 
 export type ClassNameStrategy = (
-  componentData: ComponentData,
+  componentData: ComponentData & { type?: string },
```
Comment (shaulgo) [inline packages/common/web/src/types/commonTypes.ts:27]: revert

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 5]
```diff
@@ -323,7 +329,9 @@ const createBaseComponent = ({
         PluginComponent.sizeClassName || sizeClassName,
         PluginComponent.textWrapClassName || textWrapClassName,
         PluginComponent.customClassName,
-      ]).map(strategy => strategy(this.state.componentData, theme, this.styles, isMobile));
+      ]).map(strategy =>
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:332]: revert

File: packages/viewer/web/src/getPluginViewers.jsx [hunk 6]
```diff
@@ -44,7 +45,7 @@ class PluginViewer extends PureComponent {
         : alignmentClassName(componentData, theme, styles, isMobile),
       isFunction(size)
         ? size(componentData, theme, styles, isMobile)
-        : sizeClassName(componentData, theme, styles, isMobile),
+        : sizeClassName({ ...componentData, type }, theme, styles, isMobile),
```
Comment (shaulgo) [inline packages/viewer/web/src/getPluginViewers.jsx:48]: revert

File: website/docs/ricos/ricos-api.mdx [hunk 7]
```diff
@@ -199,6 +199,20 @@ Further read in [Theming](./Theming) page.
 In order for the UGC source code to be rendered in an isolated environment, `iframeSandboxDomain` config on viewer & editor must be supplied, and be easily changed by consumer in case of DDOS on domain or subdomain
 
 
+### `textWrap`
+
+```jsx
+textWrap?: boolean;
+```
+
+Allows users to enable/disable textWrap - surround plugins (image, galleries and etc) with text.
```
Comment (shaulgo) [inline website/docs/ricos/ricos-api.mdx:208]:
```suggestion
textWrap - when a plugin is aligned left/right, text will be next the plugin. (image, galleries, ...)
```

**Review by shaulgo — APPROVED** [review]

File: packages/editor/web/src/RichContentEditor/handleBackspaceCommand.js (hunk 1 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/handleBackspaceCommand.js:34]: Done.

File: packages/viewer/web/src/RichContentViewer.tsx (hunk 2 above)
Comment (maimonav) [reply packages/viewer/web/src/RichContentViewer.tsx:79]: Done.

File: packages/common/web/src/Utils/classNameStrategies.ts (hunk 3 above)
Comment (maimonav) [reply packages/common/web/src/Utils/classNameStrategies.ts:17]: Done.

File: packages/common/web/src/types/commonTypes.ts (hunk 4 above)
Comment (maimonav) [reply packages/common/web/src/types/commonTypes.ts:27]: Done.

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx (hunk 5 above)
Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx:332]: Done.

File: packages/viewer/web/src/getPluginViewers.jsx (hunk 6 above)
Comment (maimonav) [reply packages/viewer/web/src/getPluginViewers.jsx:48]: Done.

**Pull request merged**